src/overload_ifelse.jl

```diff
@@ -0,0 +1,41 @@
+const SimpleTracer = Union{ConnectivityTracer,GradientTracer,HessianTracer}
+
+function Base.ifelse(::T, x, y) where {T<:SimpleTracer}
+    size(x) != size(y) &&
+        throw(DimensionMismatch("Outputs sizes of `ifelse` arguments don't match in size."))
+    return output_union(x, y)
+end
+
+## output union on scalar outputs
+output_union(tx::C, ty::C) where {C<:ConnectivityTracer} = C(inputs(tx) ∪ inputs(ty))
+output_union(tx::G, ty::G) where {G<:GradientTracer} = G(gradient(tx) ∪ gradient(ty))
+function output_union(tx::H, ty::H) where {H<:HessianTracer}
+    return H(gradient(tx) ∪ gradient(ty), hessian(tx) ∪ hessian(ty))
+end
+
+output_union(tx::T, y) where {T<:SimpleTracer} = tx
+output_union(x, ty::T) where {T<:SimpleTracer} = ty
+
+## output union on AbstractArray outputs
+function output_union(tx::AbstractArray{T}, ty::AbstractArray{T}) where {T<:SimpleTracer}
+    return output_union.(tx, ty)
+end
+function output_union(tx::AbstractArray{T}, y::AbstractArray) where {T<:SimpleTracer}
+    return tx
+end
+function output_union(x::AbstractArray, ty::AbstractArray{T}) where {T<:SimpleTracer}
+    return ty
+end
+
+## Overload comparisons
```

Comment:
can we add them to the binary operators pipeline, but only for global tracing?

Reply:
We should, see [here](https://github.com/adrhill/SparseConnectivityTracer.jl/pull/94#discussion_r1612311999).

Reply:
Actually not as trivial to do as I thought since the overloads of both global and dual tracers happen in the same function `overload_*_2_to_1(M, op)`.

Reply:
I split the `overload_**(M, op)` functions into separate `overload_**_dual(M, op)` functions. However, I still couldn't add the comparison operators to `operators.jl` due to the `overload_all` mechanism.

Reply:
Why not split `overload_all`?

Reply:
Can do, I didn't want to break your masterpiece. ;)

Reply:
The issue actually lies in `list_operators`, which also needs to be split.

Reply:
I'm not sure it get's more elegant than what we have now. Did you re-review?

Reply:
Yeah, basically we would need `list_operators_tracer` and `list_operators_dual`?

Reply:
Yeah basically. But then we would also have to split and duplicate our entire list in `operators.jl`. I think this solution is the simplest with the least amount of code.